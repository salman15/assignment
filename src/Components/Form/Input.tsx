import React from "react";
import styled from "styled-components";

interface InputProps {
  label?: string;
  warning?: boolean;
  warningText?: string;
  placeholder: string;
  inputValue?: string | number;
  setInputValue: Function;
  required?: boolean;
}

const Input: React.FC<InputProps> = ({
  label,
  placeholder,
  inputValue,
  setInputValue,
  warning,
  warningText,
  required,
}) => {
  return (
    <InputContainer>
      {label && (
        <Label>
          {label} {required && `*`}
        </Label>
      )}
      <InputLayout
        placeholder={placeholder}
        value={inputValue}
        onChange={(e: { target: { value: any } }) =>
          setInputValue(e.target.value)
        }
      />
      {warning && <Warning>{warningText}</Warning>}
    </InputContainer>
  );
};

export default Input;

const InputLayout = styled.input``;

const Label = styled.p`
  font-weight: bold;
  margin-bottom: 8px;
  font-size: 12px;
`;

const Warning = styled(Label)`
  color: red;
`;

const InputContainer = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-direction: column;
`;
