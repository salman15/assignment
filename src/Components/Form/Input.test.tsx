import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Input from "./Input";

describe("Test the Input component", () => {
  it("Test if Input component renders with input value", () => {
    // Act
    const { container, getByDisplayValue } = render(
      <Input
        placeholder="I'm a placeholder"
        setInputValue={() => {}}
        inputValue="I'm the input value"
      />
    );

    // Assert
    const inputElement = getByDisplayValue("I'm the input value");
    expect(container).toMatchSnapshot();
    expect(inputElement).toBeInTheDocument();
  });

  it("Test if Input component element displays the label and warning", () => {
    // Act
    const { container, getByText } = render(
      <Input
        label="I'm the label text"
        warning={true}
        warningText="I'm the warning text"
        placeholder="I'm a placeholder"
        setInputValue={() => {}}
        inputValue="I'm the input value"
      />
    );

    // Assert
    const labelElement = getByText("I'm the label text");
    const warningElement = getByText("I'm the warning text");
    expect(container).toMatchSnapshot();
    expect(labelElement).toBeInTheDocument();
    expect(warningElement).toBeInTheDocument();
  });
});
