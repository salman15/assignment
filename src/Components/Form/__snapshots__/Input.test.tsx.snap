// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Input component Test if Input component element displays the label and warning 1`] = `
<div>
  <div
    class="sc-eCssSg iuAVtC"
  >
    <p
      class="sc-gsTCUz bhSgbG"
    >
      I'm the label text
       
    </p>
    <input
      aria-label="text-input"
      autocomplete="text"
      class="sc-bdfBwQ fuywCS"
      name="label"
      placeholder="I'm a placeholder"
      type="text"
      value="I'm the input value"
    />
    <p
      class="sc-gsTCUz sc-hKgILt bhSgbG eQjPkS"
    >
      I'm the warning text
    </p>
  </div>
</div>
`;

exports[`Test the Input component Test if Input component renders with input value 1`] = `
<div>
  <div
    class="sc-eCssSg iuAVtC"
  >
    <input
      aria-label="text-input"
      autocomplete="text"
      class="sc-bdfBwQ fuywCS"
      name="label"
      placeholder="I'm a placeholder"
      type="text"
      value="I'm the input value"
    />
  </div>
</div>
`;
